game-core/src/main/java/games/strategy/engine/posted/game/pbem/PbemMessagePoster.java

```diff
@@ -102,7 +104,24 @@ public boolean post(final IDelegateHistoryWriter historyWriter, final String tit
                     (gameNameAndInfo + "\n\n" + turnSummary),
                     "TripleA " + title + ": " + currentPlayer.getName() + " round " + roundNumber,
                     saveGameFile.toPath());
-        turnSummaryRef = forumSuccess.get();
+        final AtomicBoolean success = new AtomicBoolean(false);
+        turnSummaryRef =
+            forumSuccess
+                .exceptionally(
+                    e -> {
+                      DialogBuilder.builder()
+                          .parent(null)
+                          .title("Error Posting to Forum")
+                          .errorMessage(e.getMessage())
+                          .showDialog();
+                      return null;
+                    })
+                .whenComplete((v1, v2) -> success.set(true))
+                .get();
```

Comment:
I really wish java had an `await` keyword like C# to simplify this chaining code